### PR TITLE
added async to client_api.js, added a free_size option to file.py for…

### PR DESCRIPTION
**Pull request by mrkeyan**
… portrait images, takes out white bars if  project settings of free_size = true

changes to video_wdg.py and embed_wdg.py are not my personal changes

### CODE REVIEW

File: src/tactic/ui/widget/video_wdg.py [hunk 1]
```diff
@@ -11,13 +11,12 @@
 #
 
 
-__all__ = ['VideoWdg', 'VideoJsWdg']
+__all__ = ['VideoWdg']
```
Comment (oboreo) [inline src/tactic/ui/widget/video_wdg.py]: please keep the VideoJsWdg class around

File: src/tactic/ui/widget/embed_wdg.py [hunk 2]
```diff
@@ -125,12 +125,9 @@ def get_display(my):
 
         else:
             #embed = HtmlElement.embed(src)
```
Comment (oboreo) [inline src/tactic/ui/widget/embed_wdg.py]: can you explain this change?

File: src/tactic/ui/widget/video_wdg.py [hunk 3]
```diff
@@ -399,9 +228,15 @@ def get_onload_js(my):
 
 }
 
-
 spt.video._add_events = function(player, events) {
-    return;
+    player.on("pause", function() {
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: Unnesecary.

File: src/tactic/ui/widget/video_wdg.py [hunk 4]
```diff
@@ -384,14 +217,10 @@ def get_onload_js(my):
...
         spt.dom.load_js(["video/video.js"], function() {
-            var player = videojs(
-                    video_id,
-                    {"nativeControlsForTouch": false},
-                    function() {
-                        spt.video.loaded = true;
-                        spt.video._add_events(this, events);
-                    }
-            );
+            var player = videojs(video_id, {"nativeControlsForTouch": false}, function() {
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: No change

File: src/tactic/ui/widget/video_wdg.py [hunk 5]
```diff
@@ -265,8 +123,14 @@ def get_display(my):
...
 
+            spt.video.init_player(video_id);
+            /*
+            spt.dom.load_js(["video/video.js"], function() {
+                var player = videojs(video_id, {"nativeControlsForTouch": false}, function() {
+                } );
+            });
+            */
 
-            spt.video.init_videojs(video_id);
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: No change

File: src/tactic/ui/widget/video_wdg.py [hunk 6]
```diff
@@ -265,8 +123,14 @@ def get_display(my):
 
             var video_id = bvr.video_id;
 
+            spt.video.init_player(video_id);
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: Unnessary

File: src/tactic/ui/widget/video_wdg.py [hunk 7]
```diff
@@ -297,24 +161,16 @@ def get_display(my):
...
             video.add_attr("poster", poster)
 
-        if preload == None:
-            preload = "none"
-
-        if controls == None:
-            controls = True
 
-        autoplay = False
+        if not preload:
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: No change

File: src/tactic/ui/widget/video_wdg.py [hunk 8]
```diff
@@ -197,7 +55,7 @@ def get_display(my):
         height = my.kwargs.get("height")
         preload = my.kwargs.get("preload")
         controls = my.kwargs.get("controls")
-        autoplay = my.kwargs.get("autoplay")
+
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: This may be a change

File: src/tactic/ui/widget/video_wdg.py [hunk 9]
```diff
@@ -297,24 +161,16 @@ def get_display(my):
...
-
-        if controls == None:
-            controls = True
 
-        autoplay = False
+        if not preload:
+            preload = "none"
+        video.add_attr("preload", preload)
 
-        # videojs uses a json data structre
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: Possible change

File: src/tactic/ui/widget/video_wdg.py [hunk 10]
```diff
@@ -344,13 +200,6 @@ def get_onload_js(my):
 spt.video.players = {};
 
 
-
-spt.video.get_player_el = function(el) {
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/video_wdg.py]: ?

File: src/tactic/ui/widget/embed_wdg.py [hunk 11]
```diff
@@ -66,7 +66,6 @@ def get_display(my):
         #div = DivWdg()
         #top.add(div)
         div = top
-        div.add_class("unselectable")
```
Comment (CeltonMcGrath) [inline src/tactic/ui/widget/embed_wdg.py]: add unselectable kwarg

**Pull request merged**